types/koa/index.d.ts

```diff
@@ -314,7 +315,7 @@ declare interface ContextDelegatedResponse {
     /**
      * Get/Set response body.
      */
-    body: any;
+    body: unknown;
```

Comment:
This is the clue for the whole change. With only this small change, one could type the response like this:

```js
interface MyControllerResponse {
   ...
}

export const myController: Middleware<DefaultState, { body: MyControllerResponse, response: { body: MyControllerResponse } }> = async (ctx) => {
    ...
}
```

It was pretty long though, so I extended the Middleware type below...